File: dayNine/java/Car Parking System/SimpleParkingSystem.java
public class SimpleParkingSystem {
    
    public static int[] checkParkingStatus(int[] parkingSlots) {
        return parkingSlots;
    }
    public static int parkACar(int[] parkingSlots) {
        int slot = 0;
        for(int index = 0; index < parkingSlots.length; index++){
            if(parkingSlots[index] == 0){
                parkingSlots[index] = 1;
                slot = index + 1;
                break;
            }
            
        }
        
        return slot;
    }
    public static void parkACar(int[] parkingSlots, int slot) {
         if(isAvailable(parkingSlots, slot)) {
         
            if(parkingSlots[slot - 1] == 0){
                parkingSlots[slot - 1] = 1;
            }
         }
    }
     public static boolean isAvailable(int[] parkingSlots, int slot) {
            if(parkingSlots[slot - 1] == 0){
                return true;
            }
        return false;
    }
    public static void removeACar(int[] parkingSlots, int slot){
        if(parkingSlots[slot - 1] == 1){
            parkingSlots[slot - 1] = 0;
        }
    }
    public static void displayTheSlots(int[] parkingSlots){
        for(int index = 0; index < parkingSlots.length; index++){
            int slot = index + 1;
            if(parkingSlots[index] == 0){
                System.out.println("Available slot: " + slot );
            }
            else{
               System.out.println("Occupied slot: " + slot); 
            }
        }
    }
    public static void displayAvailableSlot(int[] parkingSlots){
        for(int index = 0; index < parkingSlots.length; index++){
            int slot = index + 1;
            if(parkingSlots[index] == 0){
                System.out.println("Available slot: " + slot );
            }
        }
    }
    public static void displayOccupiedSlots(int[] parkingSlots){
        for(int index = 0; index < parkingSlots.length; index++){
            int slot = index + 1;
            if(parkingSlots[index] == 1){
               System.out.println("Occupied slot: " + slot); 
            }
        }
    }
}

File: dayNine/python/simple_parking_system.py
def check_parking_status(parking_slots):
    return parking_slots


def park_a_car(parking_slots):
    slot = 0
    for index in range(len(parking_slots)):
        if parking_slots[index] == 0:
            parking_slots[index] = 1
            slot = index + 1
            break
    return slot
    
def is_available(parking_slots, slot):
    if parking_slots[slot - 1] == 0:
        return True
    return False

def park_a_car_with_slot(parking_slots, slot):
    if is_available(parking_slots, slot):
        if parking_slots[slot - 1] == 0:
            parking_slots[slot - 1] = 1

def remove_a_car(parking_slots, slot):
    if parking_slots[slot - 1] == 1:
        parking_slots[slot - 1] = 0


def display_the_slots(parking_slots):
    for index in range(len(parking_slots)):
        slot = index + 1
        if parking_slots[index] == 0:
            print(f"Available slot: {slot}")
        else:
            print(f"Occupied slot: {slot}")


def display_available_slot(parking_slots):
    for index in range(len(parking_slots)):
        slot = index + 1
        if parking_slots[index] == 0:
            print(f"Available slot: {slot}")


def display_occupied_slots(parking_slots):
    for index in range(len(parking_slots)):
        slot = index + 1
        if parking_slots[index] == 1:
            print(f"Occupied slot: {slot}")

